web/yo/app/scripts/controllers/reportgenerator.js

```diff
@@ -407,4 +407,67 @@ angular.module('oncokbApp')
                     return 'Validate';
                 }
             }
+
+            $scope.reverseValidation = {
+                flag: false,
+                result: '',
+                validating: false
+            };
+            $scope.reverseValidateTruncating = function() {
+                $scope.reverseValidation = {
+                    flag: false,
+                    result: '',
+                    validating: true
+                };
+                DatabaseConnector.getReviewedData('geneType').then(function(response) {
+                    var geneTypes = {};
+                    var tempHugo = '';
+                    var variantCallBody = [];
+                    _.each(response, function(item) {
+                        geneTypes[item.hugoSymbol] = {
+                            oncogene: item.oncogene,
+                            tsg: item.tsg
+                        };
+                        variantCallBody.push({
+                            hugoSymbol: item.hugoSymbol
+                        });
+                    });
+                    DatabaseConnector.lookupVariants(variantCallBody).then(function(result) {
+                        var validationResult = [];
+                        // Add a validation to find tumor suppressor genes that have no truncating mutations curated
+                        _.each(result, function(alterations) {
+                            if (alterations.length > 0) {
+                                tempHugo = alterations[0].gene.hugoSymbol;
+                                if (geneTypes[tempHugo] && geneTypes[tempHugo].tsg === true) {
+                                    var isTruncating = false;
+                                    _.each(alterations, function(alteration) {
+                                        if (alteration.alteration === 'Truncating Mutations') {
+                                            isTruncating = true;
+                                        }
+                                    });
+                                    if(!isTruncating){
+                                        validationResult.push(tempHugo);
+                                    }
+                                }
+                            }
+                        });
+                        if (validationResult.length === 0) {
+                            $scope.reverseValidation.result = 'Yes! All genes passed the validation.';
+                            $scope.reverseValidation.flag = true;
+                        } else {
+                            $scope.reverseValidation.result = 'Genes that having no Truncating Mutation curated and ' +
```

Comment:
Change the message to "Tumor suppressor genes that have no Truncating Mutations curated are: "